File: Go/Slice/remove-from-slice.go
package main

import (
	"fmt"
)

func main() {
	// Create a slice of strings
	randomSlice := []string{"A", "B", "C", "D", "E", "F", "G", "H", "I", "J"}
	// Remove some elements from the slice
	newSlice := removeFromSlice(randomSlice, "A")
	fmt.Println(newSlice)
}

// Remove some value from a slice and return the slice.
func removeFromSlice(slice []string, content string) []string {
	for elementIndex, element := range slice {
		if element == content {
			slice = append(slice[:elementIndex], slice[elementIndex+1:]...)
		}
	}
	return slice
}

File: TypeScript/Slice/remove-from-slice.ts
function main(): void {
    // Create a slice
    var slice: string[] = ["a", "b", "c", "d", "e"]
    // Remove an element from the slice
    var newSlice: string[] = removeElementFromSlice(slice, "d")
    // Print the slice
    console.log(newSlice)
}

main()

// Remove an element from the provided slice and return the new slice.
function removeElementFromSlice(slice: string[], element: string): string[] {
    var index: number = slice.indexOf(element)
    if (index > -1) {
        slice.splice(index, 1)
    }
    return slice
}